libgalois/include/katana/PropertyGraph.h

```diff
@@ -95,57 +95,7 @@ class KATANA_EXPORT PropertyGraph {
 
   PGViewCache pg_view_cache_;
 
-  // Keep partition_metadata, master_nodes, mirror_nodes out of the public interface,
-  // while allowing Distribution to read/write it for RDG
-
-  friend class Distribution;
-  const tsuba::PartitionMetadata& partition_metadata() const {
-    return rdg_.part_metadata();
-  }
-
-  void set_partition_metadata(const tsuba::PartitionMetadata& meta) {
-    rdg_.set_part_metadata(meta);
-  }
-
-  void update_rdg_metadata(const std::string& part_policy, uint32_t num_hosts) {
-    rdg_.set_view_name(fmt::format("rdg-{}-part{}", part_policy, num_hosts));
-  }
-
-  /// Per-host vector of master nodes
-  ///
-  /// master_nodes()[this_host].empty() is true
-  /// master_nodes()[host_i][x] contains LocalNodeID of masters
-  //    for which host_i has a mirror
-  const std::vector<std::shared_ptr<arrow::ChunkedArray>>& master_nodes()
-      const {
-    return rdg_.master_nodes();
-  }
-  void set_master_nodes(std::vector<std::shared_ptr<arrow::ChunkedArray>>&& a) {
-    rdg_.set_master_nodes(std::move(a));
-  }
-
-  /// Per-host vector of mirror nodes
-  ///
-  /// mirror_nodes()[this_host].empty() is true
-  /// mirror_nodes()[host_i][x] contains LocalNodeID of mirrors
-  ///   that have a master on host_i
-  const std::vector<std::shared_ptr<arrow::ChunkedArray>>& mirror_nodes()
-      const {
-    return rdg_.mirror_nodes();
-  }
-  void set_mirror_nodes(std::vector<std::shared_ptr<arrow::ChunkedArray>>&& a) {
-    rdg_.set_mirror_nodes(std::move(a));
-  }
-
-  /// Return the node property table for local nodes
-  const std::shared_ptr<arrow::Table>& node_properties() const {
-    return rdg_.node_properties();
-  }
-
-  /// Return the edge property table for local edges
-  const std::shared_ptr<arrow::Table>& edge_properties() const {
-    return rdg_.edge_properties();
-  }
+  friend class PropertyGraphRetractor;
```

Comment:
I'm going to insist on a better name. More importantly, we should define the purpose of this class clearly. 

A less urgent but important issue at hand is that PropertyGraph should be made unmoveable so that all parties keep a unique_ptr<PropertyGraph> or shared_ptr<> if that makes sense in certain scenarios.

Reply:
The purpose of this class is to allow:
 * access to data that is only useful when storing partitioned graphs
 * access to methods that will leave PropertyGraph in a state where it's other invariants don't hold.

Do you have any suggestions for a better name?

Reply:
`PropertyGraphDistributedMetaDataManager`? `BandaidForBadLeakyAbstraction`? ;)

Reply:
> A less urgent but important issue at hand is that PropertyGraph should be made unmoveable so that all parties keep a unique_ptr or shared_ptr<> if that makes sense in certain scenarios.

Maybe but I'm not going to address that here.

Reply:
I have not reviewed the code yet, but OMG, I thought PropertyGraphRetractor was the most brilliant name I've ever heard.  I loved the TV show MASH when I was a kid, so I know that "nurse, retractor" means its time to get into the guts.  Perfect.